File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> sizes) {
    int x = 0;
    int y = 0;
    for (auto &size : sizes) {
        x = max(x, min(size[0], size[1]));
        y = max(y, max(size[0], size[1]));
    }
    
    int answer = x * y;
    return answer;
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    max_x = 0
    max_y = 0
    
    for cur_x, cur_y in map(sorted, sizes):
        max_x = max(max_x, cur_x)
        max_y = max(max_y, cur_y)
    
    answer = max_x * max_y
    return answer